src/test/ECSTest.cpp

```diff
@@ -234,3 +281,71 @@ TEST_F(ECSTest, partentChild) {
 	EXPECT_EQ(metadata[1].get().children[0], 3);
 	EXPECT_EQ(metadata[2].get().children[0], 4);
 }
+
+TEST_F(ECSTest, persistent) {
```

Comment:
Ik weet dat we tests eigenlijk niet horen te controleren op code standaard, maar ik heb liever dat elke test maar één ding test, en in deze test controleer je ook een hoop andere properties. Mij lijkt dat alleen de Metadata.name genoeg is om te kijken welke componenten nog bestaan voor/na de `mgr.delete_all_components();` aanroep. Ik neem aan dat er al een andere test is binnen de `ECSTest` suite die netjes controleert of `Metadata` en `Transform` beide juist geïnitialiseerd zijn dus dat hoeft niet in elke test tenzij dit noodzakelijk is (dan zou ik ook `ASSERT_*` ipv `EXPECT_*` verwachten).